### DISCUSSION TITLE
Can EPF be used on a game trigger?

### DISCUSSION THREAD

**Discussion by PapaReap** [Arkensor/EnfusionPersistenceFramework#42] [Q&A] [answered]
We're really enjoying this mod, great work.

I  have a custom trigger that I've created to spawn units, based off of SCR_BaseTriggerEntity.
I was wondering if EPF can keep track of trigger states?
I was also wondering if EPF can also persist the units spawned with the trigger, they are SCR_AIGroup?
So on restart of mission, a trigger that has already been activated, will not activate again after restart.
I've read through all that I can make sense of and have had no luck to this point.
I've tried adding the EPF_PersistenceComponent as a child to my trigger, no luck.

Thanks,
PapaReap

**Comment by Arkensor** [comment, marked as answer]
Good question, at the very least I have no default trigger persistence implemented. It should be possible to save and restore the trigger with additional script logic for them. The AI group could also be persisted, right now I just did not do that.
However, all of this might be supported in vanilla soon-ish as I work on applying some of the mod ideas from this project into the game itself. No promises yet, but this looks like a valid use case I will likely want to consider.

**Reply by PapaReap** [reply]
Thanks for the reply, saves me further tinkering with my limited knowledge on your mod to get it to work.

Our group has been using EPF with our custom PVE conflict mission we've been running over the past several weeks.
It's doing what we need, mainly saving vehicle info's, we love the fact that our destroyed vehicles come back destroyed, gives the feel of real battlefield. 

We're not using the save player position & loadouts because we use custom loadouts and spawn at various bases (I don't think you have support for anything like this atm)

The triggers I've made, for the most part are not a issue not persisting, I want them spawning each session anyways. I have them running random renegade patrols spawned at random positions and times retaking bases and just generating general havoc. They basically run on a die/spawn loop anyways all session long. Same for my random task triggers, they can repeat also, as there random times and placements also.

Just a few triggers, where it's obvious that they shouldn't repeat is the only issue. Not a deal breaker by any means, I'll come up with a work around.

Again, thanks for the mod. It really adds a ton of enjoyment to our group.

PapaReap

PS I just had a idea that I might follow up on. Perhaps, I can create a object in the world that would persist. Make a condition on my trigger that the object has to be alive, along with the other activation methods, maybe I could kill the persistent object,  On restart, object would be dead and trigger couldn't fire. IDK, just spit balling.
